data-contracts/README.md

```diff
@@ -0,0 +1,439 @@
+
+This command may open a browser window asking you to authenticate to Confluent Cloud. Once that's complete, the result of
+`confluent organization list` is queried by `jq` to extract the `id` of the current organization to which you are authenticated:
+
+```shell
+export TF_VAR_org_id=$(confluent organization list -o json | jq -c -r '.[] | select(.is_current)' | jq '.id')
+```
+
+Now, we are ready to initialize the terraform environment, create a `plan` and `apply` said plan to create CC assets:
+
+```shell
+terraform init
+terraform plan -out "tfplan"
+terraform apply "tfplan"
+```
+
+Let's have a look at what we've created. We now have a Confluent Cloud environment, which includes a Kafka cluster and Data Governance platform with Confluent Schema Registry. Within the Kafka cluster, we create a Kafka topic named `membership-avro`.
+
+```terraform
+resource "confluent_kafka_topic" "membership_avro" {
+  topic_name = "membership-avro"
+  kafka_cluster {
+    id = confluent_kafka_cluster.kafka_cluster.id
+  }
+  rest_endpoint = confluent_kafka_cluster.kafka_cluster.rest_endpoint
+...
+  partitions_count = 10
+...
+}
+```
+
+Then we configure the subject containing the value of events for this topic using `confluent_subject_config`. This tells the Schema Registry that `membership-avro-value` schemas will be `BACKWARD` compatible and defines an attribute called `major_version` to be used as the `compatibilityGroup`.
+
+```terraform
+resource "confluent_subject_config" "membership_value_avro" {
+  subject_name = "${confluent_kafka_topic.membership_avro.topic_name}-value"
+
+  schema_registry_cluster {
+    id = data.confluent_schema_registry_cluster.advanced.id
+  }
+  rest_endpoint = data.confluent_schema_registry_cluster.advanced.rest_endpoint
+...
+  compatibility_level = "BACKWARD"
+  compatibility_group = "major_version"
+...  
+}
+```
+
+For more on these data contract configuration enhancements, refer to the CC docs on [data contracts](https://docs.confluent.io/cloud/current/sr/fundamentals/data-contracts.html#configuration-enhancements). 
+
+
+#### Prepare Client Properties
+
+This demo has tailored the output of `terraform apply` to return the properties needed to connect to Confluent Cloud. The command below will 
+reformat the names of those properties into the names used in Kafka Client configurations, then export those outputs to a properties file 
+in our project:
+
+```shell
+terraform output -json | \
+  jq -r 'to_entries | map( {key: .key|tostring|split("_")|join("."), value: .value} ) | map("\(.key)=\(.value.value)")' | while read -r line ; do echo "$line"; \
+  done > ../shared/src/main/resources/confluent.properties
+```
+
+All Kafka Client code in this project loads connection properties form `shared/src/main/resources/confluent.properties`. For an example of this 
+properties file, see [confluent.properties.orig](shared/src/main/resources/confluent.properties.orig).
```

Comment:
full URL so that it works on CD
```suggestion
properties file, see [confluent.properties.orig](https://raw.githubusercontent.com/confluentinc/tutorials/refs/heads/master/data-contracts/shared/src/main/resources/confluent.properties.orig).
```